### PR TITLE
Global search improvements

### PR DESCRIPTION
**Pull request by bastianjoel**
resolves part of #2738 

Note that this needs the latest `search.yml` changes of the search service.

### CODE REVIEW

File: client/src/app/site/modules/global-headbar/components/global-search/global-search.component.html [hunk 1]
```diff
@@ -89,4 +83,5 @@ <h3 *ngIf="noResults" class="search-no-results">{{ 'No results found' | translat
             <mat-spinner></mat-spinner>
         </div>
     </ng-container>
+    <div style="display: none">Results {{ filteredResultCount }}/{{ resultCount }}</div>
```
Comment (bastianjoel) [inline client/src/app/site/modules/global-headbar/components/global-search/global-search.component.html:86]: This is intended.

**Comment by bastianjoel** [comment]
For some reason the changes from c0cef8cfab96ea755bdcb0c7d69770c0c64b8556 are currently not displayed here.

File: client/src/app/site/services/global-search/global-search.service.ts [hunk 2]
```diff
@@ -48,6 +48,18 @@ export class GlobalSearchService {
...
+            const firstName = content.first_name?.trim() || ``;
+            const lastName = content.last_name?.trim() || ``;
+            const userName = content.username?.trim() || ``;
+            const name = firstName || lastName ? `${firstName} ${lastName}` : userName;
+            return `${content.title?.trim() || ``} ${name?.trim()}`.trim() || ``;
+        }
+
+        return content.title || content.name;
+    }
+
```
Comment (reiterl) [inline client/src/app/site/services/global-search/global-search.service.ts:62]: Why did you move this code?

**Review by reiterl — COMMENTED** [review]
I'm not sure, why you moved one code snippet. Looks else okay so far.

File: client/src/app/site/services/global-search/global-search.service.ts (hunk 2 above)
Comment (bastianjoel) [reply client/src/app/site/services/global-search/global-search.service.ts:62]: We have the convention that public methods should be defined above private or protected ones. This is just cleanup.

**Review by Elblinator — APPROVED** [review]
